### DISCUSSION TITLE
Micronaut 4 estimated release date?

### DISCUSSION THREAD

**Discussion by donbeave** [micronaut-projects/micronaut-core#7827] [General] [answered]
Hi, guys!

I want to ask whether you have any estimated date for releasing Micronaut 4, something like a very rough date. Is it planning on this year, on it's clear it will be the next year?

Sorry if I ask a question that is probably already somewhere replied.

Best regards,
  Alexey.

**Comment by graemerocher** [comment, marked as answer]
by the end of the year

**Reply by dstepanov** [reply]
Is there something you are waiting for?

**Reply by donbeave** [reply]
Mostly only upgrade dependencies to the latest versions. In my case, I'm interested in:
1) graphql-java: https://github.com/micronaut-projects/micronaut-graphql/pull/247#issuecomment-1202343945
2) Java 17 baseline: https://github.com/micronaut-projects/micronaut-build/issues/323, https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/367
3) gradle docker plugin: https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/363

**Comment by corporatepiyush** [comment]
Still haven't released yet. Any new deadline for 4.0 ? I really wish to run it on Java 17.

**Reply by graemerocher** [reply]
End of the month

**Reply by alvarosanchez** [reply]
@corporatepiyush you can run Micronaut v3 on Java 17 already.

**Reply by subash89** [reply]
Noticed the RC4 released couple days ago. Should we expect GA release this month ?

**Reply by alvarosanchez** [reply]
The RC4 version you mention is of Micronaut Core only, which now gets versioned independently of what was formerly known as the Micronaut version, which is now Micronaut Platform: https://github.com/micronaut-projects/micronaut-platform/releases

That said, we are definitely aiming towards a Micronaut Platform release candidate very soon

**Reply by subash89** [reply]
@alvarosanchez  ok I meant the one that carries the actual code/building blocks for a micro service, which apparently is the platform and its related components. Do you have an estimated release date for that ?

**Reply by sdelamo** [reply]
We plan to release GA this week.

**Reply by corporatepiyush** [reply]
Call me greedy but I wish to know if this release also can be Virtual Threads ready  ? JDK 21 is releasing in couple of months.

**Reply by yawkat** [reply]
@corporatepiyush 4.0 introduces virtual thread support: #8180

**Reply by corporatepiyush** [reply]
Yay ! Thanks

**Reply by subash89** [reply]
First off congratulations on the 4.0 release of micronaut core.  
I see micronaut-security still on Ms, not 4.0 released  yet.  I assume all micronaut projects have to be on 4.0 for them to work together since it's a major version change ?

**Reply by yawkat** [reply]
@subash89 yes, module releases will be done in the coming days. 4.0 will be considered released once micronaut-platform is on 4.0.
